### PR TITLE
Api fuzzer extensions to support simulating traffic congestion

### PR DESCRIPTION
**Pull request by Vignesh2208**
Simulates traffic congestion by capping the amount of readable/writable bytes through a channel at any given moment in time.

### CODE REVIEW

File: test/core/end2end/fuzzers/api_fuzzer.cc [hunk 1]
```diff
@@ -43,6 +43,14 @@
 #include "test/core/end2end/fuzzers/api_fuzzer.pb.h"
 #include "test/core/util/passthru_endpoint.h"
 
+#define MAX_ADVANCE_TIME_MICROS 100000 // 100ms
```
Comment (ctiller) [inline test/core/end2end/fuzzers/api_fuzzer.cc:46]: How about 1 year?

File: test/core/end2end/fuzzers/api_fuzzer.cc [hunk 2]
```diff
@@ -43,6 +43,14 @@
...
+// Applicable when simulating channel actions. Prevents overflows.
+#define MAX_WAIT_MS 1000
```
Comment (ctiller) [inline test/core/end2end/fuzzers/api_fuzzer.cc:48]: again, maybe a year?

File: test/core/end2end/fuzzers/api_fuzzer.cc [hunk 3]
```diff
@@ -43,6 +43,14 @@
...
+// Applicable when simulating channel actions. Prevents overflows.
+#define MAX_ADD_N_READABLE_BYTES 1024
```
Comment (ctiller) [inline test/core/end2end/fuzzers/api_fuzzer.cc:50]: maybe a few gigabytes?

File: test/core/end2end/fuzzers/api_fuzzer.cc [hunk 4]
```diff
@@ -118,7 +128,7 @@ void my_resolve_address(const char* addr, const char* /*default_port*/,
   r->on_done = on_done;
   r->addrs = addrs;
   grpc_timer_init(
-      &r->timer, GPR_MS_PER_SEC + grpc_core::ExecCtx::Get()->Now(),
+      &r->timer, MAX_ADVANCE_TIME_MICROS/1000 + grpc_core::ExecCtx::Get()->Now(),
```
Comment (ctiller) [inline test/core/end2end/fuzzers/api_fuzzer.cc:131]: MS_PER_SEC is probably better here... it's some arbitrary hardcoded constant that we should replace with a fuzzer selected value at some point

File: test/core/end2end/fuzzers/api_fuzzer.cc (hunk 1 above)
Comment (Vignesh2208) [reply test/core/end2end/fuzzers/api_fuzzer.cc]: Changed.

File: test/core/end2end/fuzzers/api_fuzzer.cc (hunk 2 above)
Comment (Vignesh2208) [reply test/core/end2end/fuzzers/api_fuzzer.cc]: Changed.

File: test/core/end2end/fuzzers/api_fuzzer.cc (hunk 3 above)
Comment (Vignesh2208) [reply test/core/end2end/fuzzers/api_fuzzer.cc]: Changed.

File: test/core/end2end/fuzzers/api_fuzzer.cc (hunk 4 above)
Comment (Vignesh2208) [reply test/core/end2end/fuzzers/api_fuzzer.cc]: Restored.

**Review by ctiller — APPROVED** [review]

**Comment by gnossen** [comment]
@Vignesh2208 It looks like this is causing Windows RBE failures:

```
test/core/util/passthru_endpoint.cc(206): error C2065: 'ssize_t': undeclared identifier
test/core/util/passthru_endpoint.cc(206): error C2146: syntax error: missing ';' before identifier 'split_length'
test/core/util/passthru_endpoint.cc(206): error C2065: 'split_length': undeclared identifier
test/core/util/passthru_endpoint.cc(208): error C2065: 'split_length': undeclared identifier
test/core/util/passthru_endpoint.cc(211): error C2065: 'split_length': undeclared identifier
test/core/util/passthru_endpoint.cc(218): error C2065: 'split_length': undeclared identifier
test/core/util/passthru_endpoint.cc(221): error C2065: 'split_length': undeclared identifier
test/core/util/passthru_endpoint.cc(233): error C2065: 'split_length': undeclared identifier
test/core/util/passthru_endpoint.cc(234): error C2065: 'split_length': undeclared identifier
test/core/util/passthru_endpoint.cc(237): error C2065: 'split_length': undeclared identifier
test/core/util/passthru_endpoint.cc(238): error C2065: 'split_length': undeclared identifier
ERROR: T:/src/github/grpc/test/core/util/BUILD:33:16: Compiling test/core/util/passthru_endpoint.cc failed: (Exit 2): cl.exe failed: error executing command
  cd T:/_bazel_output/lndsuir7/execroot/com_github_grpc_grpc
  SET BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1
    SET GRPC_BAZEL_RUNTIME=1
    SET INCLUDE=C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE;C:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt;C:\Program Files (x86)\Windows Kits\8.1\include\shared;C:\Program Files (x86)\Windows Kits\8.1\include\um;C:\Program Files (x86)\Windows Kits\8.1\include\winrt;
    SET PATH=C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64;C:\Windows\Microsoft.NET\Framework64\v4.0.30319;C:\Windows\Microsoft.NET\Framework64\;C:\Program Files (x86)\Windows Kits\8.1\bin\x64;C:\Program Files (x86)\Windows Kits\8.1\bin\x86;;C:\Windows\system32
    SET PWD=/proc/self/cwd
    SET RUNFILES_MANIFEST_ONLY=1
    SET TEMP=C:\Users\ContainerAdministrator\AppData\Local\Temp
    SET TMP=C:\Users\ContainerAdministrator\AppData\Local\Temp
  C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe /nologo /DCOMPILER_MSVC /DNOMINMAX /D_WIN32_WINNT=0x0601 /D_CRT_SECURE_NO_DEPRECATE /D_CRT_SECURE_NO_WARNINGS /bigobj /Zm500 /EHsc /wd4351 /wd4291 /wd4250 /wd4996 /I. /Ibazel-out/x64_windows-opt/bin /Iexternal/upb /Ibazel-out/x64_windows-opt/bin/external/upb /Iexternal/com_google_protobuf /Ibazel-out/x64_windows-opt/bin/external/com_google_protobuf /Iexternal/com_google_absl /Ibazel-out/x64_windows-opt/bin/external/com_google_absl /Iexternal/zlib /Ibazel-out/x64_windows-opt/bin/external/zlib /Iexternal/com_github_cares_cares /Ibazel-out/x64_windows-opt/bin/external/com_github_cares_cares /Itest/core/util/include /Ibazel-out/x64_windows-opt/bin/test/core/util/include /Itest/core/util/src/core/ext/upb-generated /Ibazel-out/x64_windows-opt/bin/test/core/util/src/core/ext/upb-generated /Itest/core/util/src/core/ext/upbdefs-generated /Ibazel-out/x64_windows-opt/bin/test/core/util/src/core/ext/upbdefs-generated /Iinclude /Ibazel-out/x64_windows-opt/bin/include /Isrc/core/ext/upb-generated /Ibazel-out/x64_windows-opt/bin/src/core/ext/upb-generated /Isrc/core/ext/upbdefs-generated /Ibazel-out/x64_windows-opt/bin/src/core/ext/upbdefs-generated /Iexternal/zlib /Ibazel-out/x64_windows-opt/bin/external/zlib /Ithird_party/xxhash /Ibazel-out/x64_windows-opt/bin/third_party/xxhash /Ithird_party/address_sorting/include /Ibazel-out/x64_windows-opt/bin/third_party/address_sorting/include /Iexternal/com_github_cares_cares /Ibazel-out/x64_windows-opt/bin/external/com_github_cares_cares /DCARES_STATICLIB /DGRPC_PORT_ISOLATED_RUNTIME=1 /showIncludes /MD /O2 /Oy- /DNDEBUG /wd4117 -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" /Gy /Gw -DGRPC_BAZEL_BUILD /Fobazel-out/x64_windows-opt/bin/test/core/util/_objs/grpc_test_util_base/passthru_endpoint.obj /c test/core/util/passthru_endpoint.cc
Execution platform: //third_party/toolchains:rbe_windows_2019
```

https://source.cloud.google.com/results/invocations/d330fe28-fe0f-4bb2-906d-f70452f60863/log

**Pull request merged**